app/views/courses/index.html.erb

```diff
@@ -1,32 +1,39 @@
 <p id="notice"><%= notice %></p>
+<%= render 'courses/search_form.html.erb' %>
 
-<h1>Courses</h1>
-
-<table>
-  <thead>
-    <tr>
-      <th colspan="3"></th>
-    </tr>
-  </thead>
-
-  <tbody>
-    <% @courses.each do |course| %>
-   <div>
-     <div>
-      <%= course.name %>
-     </div>
-     <div>
-     <%= course.description %>
-     </div>
-
-     <%= link_to 'Show', course %>
-     <%= link_to 'Edit', edit_course_path(course) %>
-     <%= link_to 'Destroy', course, method: :delete, data: { confirm: 'Are you sure?' } %>
-   </div>
-    <% end %>
-  </tbody>
-</table>
-
-<br>
-
-<%= link_to 'New Course', new_course_path %>
+<div class="table-responsive">
+  <table class="table table-striped table-hover">
+    <thead>
+      <tr>
+        <th scope="col"><%= sort_link(@q, :name, 'Course Name', default_order: :asc) %> </th>
+        <th scope="col"><%= sort_link(@q, :description, 'Description', default_order: :asc) %>  </th>
+        <th scope="col"><%= sort_link(@q, :status, 'Status', default_order: :asc) %> </th>
+        <th scope="col"><%= sort_link(@q, :students_count, 'Students count', default_order: :asc) %> </th>
+        <th scope="col">Links</th>
+      </tr>
+    </thead>
+    <tbody>
+      <% @courses.each do |course| %>
+        <tr>
+          <td><%= course.name.truncate(40)%></td>
+          <td><%=course.description.truncate(40) %></td>
```

Comment:
Use decorators: https://medium.com/@rwubakwanayoolivier/using-the-decorator-pattern-in-ruby-on-rails-dd66ac17a1d0